### ISSUE TITLE
A11y: Widget with role of listbox must have no more than one tabbable element in DropdownV2 component 

### ISSUE THREAD

**Issue by icemanlily** [IBM/carbon-components-react#1851]
===== ENVIRONMENT =====
IBM CI 162 Checkpoint Violation: 2.1.1 Keyboard
IBM Guidance: https://www.ibm.com/able/guidelines/ci162/keyboard.html
Test Tool Type: DAP

===== DEFECT DESCRIPTION =====
The rule detected a WAI-ARIA widget with more than one tab stop.

The primary keyboard navigation convention uses Tab and Shift+Tab key commands to move focus from one UI component to another. Other keys (primarily the arrow keys) move focus within components comprised of multiple focusable elements. Authors must follow this convention and provide no more than one tab stop per “widget” (providing keyboard focus).

WAI-ARIA (ARIA) widgets are custom user interface objects given an explicit ARIA role that matches its intended purpose. These “roles” can be used to build standalone user interface widgets, or they might be used to create individual parts of a composite widget (e.g., role="button", role="checkbox", role="combobox", etc.).

Note: This rule checks for the most common widgets which require a tab stop.

=====STEPS TO REPRODUCE =====

1. Go to https://cloud.ibm.com/login?redirect=%2F

===== EXPECTED RESULT =====
https://aat.w3ibm.mybluemix.net/token/a80a8d09-7f5a-480a-adcc-7e22f77482b0/c19cc7e4-e2e6-4475-89d7-533c2bc34979/archives/2018OctDeploy//doc/w3/help/en-US/idhi_accessibility_check_g1178.html

![image](https://user-images.githubusercontent.com/6970471/52547811-11cef200-2e05-11e9-9453-43259a882592.png)

**Comment by dakahn** [comment]
Yeah, that component has problems. It's implemented aria wise as a menu which means it should support the full gamut of [OS level menu controls](https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus), but the menu items _should not_ be tabbable. **Tab is reserved for moving between menus.** 

So there are two ways to address this:
1) remove the menu items from the tab order silencing the automated testing error
2) remove the aria-menu attribute and let the drop down act as a list of links with tab stops

The second solution falls more in line with expected user behavior. We've implemented a tab stop keyboard interaction pattern in the experimental UI-Shell's dropdowns and those are so far testing well.

**Comment by dakahn** [comment]
adding more to this -- can't reproduce the error with the DropdownV2. Do you know which version you're using?

**Comment by icemanlily** [comment]
@dakahn The carbon-components-react version is 6.58.0

**Comment by icemanlily** [comment]
@dakahn I can reproduce the error with the DropdownV2.
![image](https://user-images.githubusercontent.com/6970471/53309215-9cbdeb00-38e1-11e9-9579-5947be59b96a.png)

**Comment by icemanlily** [comment]
@dakahn I tried it and remove tabindex="0" in listbox.js can fix this issue.
But I'm not sure if it will new problem.

**Comment by dakahn** [comment]
Getting back to this -- I am unable to reproduce your DAP error in either your linked app or in our storybook. Do me a solid and click the hamburger in the top right of the DAP panel and tell me what rule sets you're testing against. For reference here's mine:

<img width="738" alt="Screen Shot 2019-03-11 at 5 41 05 PM" src="https://user-images.githubusercontent.com/40970507/54162939-5a3df780-4425-11e9-9b9a-461d289881ab.png">

**Comment by icemanlily** [comment]
@dakahn I checked IBM Accessibility BETA in the rulesets. 
![image](https://user-images.githubusercontent.com/6970471/54168707-e1817e80-44aa-11e9-94b5-c625b959cbd4.png)

**Comment by dakahn** [comment]
Yeah, cool -- that's what I thought was up. I've pinged someone who might know about the specific differences in the latest rule set vs the BETA rule set to figure out how to move forward.

**Comment by dakahn** [comment]
I'm going to close this as resolved. The error was from the experimental rule set. Happy to open this up again in the future if we need to. 👍🏽

**Comment by icemanlily** [comment]
@dakahn I think this is an issue. We must type tab key twice to focus on the dropdown list. When we focus on the dropdown list first time, we can't operate it. Then type the tab key, We can see the focus is not changed and we can type space to operate the dropdown. 
I think it is not correct. The widget should be focus on one time and no more than one tabbable element in it.

**Comment by dakahn** [comment]
@icemanlily can you open a separate issue for that and i'll look into what you're saying. 👍🏽

**Comment by icemanlily** [comment]
@dakahn Open the separate issue. #2081
